akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSourceSpec.scala

```diff
@@ -292,6 +292,17 @@ class QueueSourceSpec extends StreamSpec {
       sourceQueue2.watchCompletion().isCompleted should ===(false)
     }
 
+    "return true on Enqueuing the element successfully when reject elements on back-pressure" in {
+      val (source, probe) = Source.queue[Int](0, OverflowStrategy.backpressure).toMat(TestSink.probe)(Keep.both).run()
+      val f = source.offer(42)
+      val ex = source.offer(43).failed.futureValue
+      ex shouldBe a[IllegalStateException]
+      ex.getMessage should include("have to wait")
+      probe.requestNext() should ===(42)
+      f.futureValue should ===(QueueOfferResult.Enqueued)
+      assert(QueueOfferResult.Enqueued.isEnqueued)
+    }
```

Comment:
Nothing is changed about the actual operators, so remove this separate test. I suggested to change the `assertSuccess` but instead, add simple unit tests for each of the possible QueueOfferResults and the returned `isEnqueued` value at the end of this test.

Something like:
```scala
"QueueOfferResult" should {
  "be enqueued for Enqueued" in {
    Enqueued.isEnqueued should === (true)
  }
 "not be enqueued for Dropped" in {
    Dropped.isEnqueued should === (false)
  }
  ... add cases verifying false for the other non enqueued types ...
}
```